chainer/utils/meta.py

```diff
@@ -0,0 +1,79 @@
+import warnings
+
+import six
+
+
+def final(*args, **kwargs):
+    """Decorator to declare a method final.
+
+    By default, :class:`TypeError` is raised when the decorated method is being
+    overridden.
+
+    The class in which the decorated method is defined must inherit from a
+    base class returned by :meth:`~chainer.utils.enable_final`.
+
+    Args:
+        action: Specifies what happens when the decorated method is being
+            overridden. If `'error'`, :class:`TypeError` is raised. If
+            :class:`DeprecationWarning`, a deprecation warning is raised.
+    """
+    def wrap(f, action='error'):
+        assert callable(f)
+        f.__is_final = (action,)
```

Comment:
How about naming it `f.__override_action`?

Reply:
Fixed.